### PR TITLE
BUG 1828704: Return invalid configuration errors when creating NICs with invalid config

### PR DESCRIPTION
**Pull request by JoelSpeed**
**What this PR does / why we need it**:

If the configuration for creating the NIC for the VM is invalid, return an InvalidConfiguration error from Create so that the Machine controller marks the machine as failed

There are definitely more occasions where we should be return invalid configuration here as well, but I'll leave those for another PR

### CODE REVIEW

File: pkg/cloud/azure/actuators/machine/actuator_test.go [hunk 1]
```diff
@@ -792,3 +793,100 @@ func TestStatusCodeBasedCreationErrors(t *testing.T) {
...
+					}
+				},
+				// use fake recorder and store an event into one item long buffer for subsequent check
+				EventRecorder: &record.FakeRecorder{
+					Events: eventsChannel,
+				},
+			})
+
+			// networkSvc.EXPECT().CreateOrUpdate(gomock.Any(), gomock.Any()).Return(nil).Times(1)
+			// vmSvc.EXPECT().CreateOrUpdate(gomock.Any(), gomock.Any()).Return(nil).Times(1)
```
Comment (elmiko) [inline pkg/cloud/azure/actuators/machine/actuator_test.go:864]: are these unused?

Comment (JoelSpeed) [reply pkg/cloud/azure/actuators/machine/actuator_test.go]: Good catch, removed

**Pull request merged**